examples/systemcalls/systemcalls.c

```diff
@@ -58,10 +111,60 @@ bool do_exec(int count, ...)
  *   as second argument to the execv() command.
  *
 */
+    int status;
+    pid_t pid;
+    pid = fork ();
+    //handle if fork failed
+    if (pid == -1)
+    {
+        perror("Fork");
+        va_end(args);
+        return false;
+    }
+    //if child process
+    else if (pid == 0) 
+    {
+        if(execv (command[0], (command)) == -1)
+        {
+            //if exec returns, that means it failed
+            perror("Exec");
+            va_end(args);
+            exit(-1);    
+        }
 
-    va_end(args);
-
-    return true;
+    }
+    //if parent process
+    else
+    {
+        if (waitpid (pid, &status, 0) == -1)
+        {
+            //if there's an issue with the invocation of the the waitpid function
+            perror("Wait");
+            va_end(args);
+            return false;
+        }
+        //if the child process returned normally (did not exit by unnatural means like a signal)
+        else if (WIFEXITED (status))
+        {
+            //check exit status of the child
+            int ret_status = WEXITSTATUS(status);
+            if(ret_status!=0)
+            {
+                printf("Command exited with non-zero status\n");
+                va_end(args);
+                return false;
```

Comment:
@GuruprashanthKrishnakumar rather than duplicating these inline cleanup/return statements, think about a design where you just set a status which is returned in one place after cleanup in one path.